src/components/accessibility-disability-simulator/accessibility-disability-simulator.js

```diff
@@ -0,0 +1,229 @@
+			'd2l-offscreen': this.disabilityType === DISABILITY_TYPES.noVision
+		};
+		const wrapperStyles = {
+			...(this.disabilityType === DISABILITY_TYPES.lowVision && { filter: `blur(${this.blurLevel / 20}px)` })
+		};
+		return html`
+			${this._renderColorblindFilters()}
+			${this._renderControls()}
+			${this._renderAlert()}
+
+			<div class="${classMap(wrapperClasses)}" style="${styleMap(wrapperStyles)}">
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		if (changedProperties.has('blurLevel')) {
+			this.blurLevel = Math.max(Math.min(this.blurLevel, 100), 1);
+		}
+	}
+
+	_localize(key) {
+		return this.localize(`components:accessibilityDisabilitySimulator:${key}`);
+	}
+
+	_onDisabilityTypeChanged(e) {
+		this.disabilityType = e.target.value;
+	}
+
+	_onSliderChanged(e) {
+		this.blurLevel = e.target.value;
+	}
+
+	_renderAlert() {
+		if (this.hideAlert || this.disabilityType !== DISABILITY_TYPES.noVision) return;
+
+		return html`
+			<d2l-alert class="alert-message" no-padding>
+				<span class="d2l-body-compact">${this._localize('screenreaderAndTabAlert')}</span>
+			</d2l-alert>
+		`;
+	}
+
+	_renderColorblindFilters() {
+		return html`
+			<svg xmlns="http://www.w3.org/2000/svg" class="svg-filter">
+				<filter id="tritanopia" color-interpolation-filters="linearRGB">
+					<feColorMatrix type="matrix" in="SourceGraphic" result="ProjectionOnPlane1" values="
+						1.01354, 0.14268, -0.15622, 0, 0
+						-0.01181, 0.87561, 0.13619, 0, 0
+						0.07707, 0.81208, 0.11085, 0, 0
+						7.92482, -5.66475, -2.26007, 1, -0.2"/>
+
+					<feComponentTransfer in="ProjectionOnPlane1" result="ProjectionOnPlane1">
+						<feFuncA type="discrete" tableValues="0 0 0 0 1"/>
+					</feComponentTransfer>
+
+					<feColorMatrix type="matrix" in="SourceGraphic" result="ProjectionOnPlane2" values="
+						0.93337, 0.19999, -0.13336, 0, 0
+						0.05809, 0.82565, 0.11626, 0, 0
+						-0.37923, 1.13825, 0.24098, 0, 0
+						0,0,0,1,0"
+					/>
+
+					<feBlend in="ProjectionOnPlane1" in2="ProjectionOnPlane2" mode="normal"/>
+				</filter>
+
+				<filter id="deuteranopia" color-interpolation-filters="linearRGB">
+					<feColorMatrix type="matrix" in="SourceGraphic" values="
+						0.29031, 0.70969, 0, 0, 0
+						0.29031, 0.70969, 0, 0, 0
+						-0.02197, 0.02197, 1, 0, 0
+						0, 0, 0, 1, 0"
+					/>
+				</filter>
+
+				<filter id="protanopia" color-interpolation-filters="linearRGB">
+					<feColorMatrix type="matrix" in="SourceGraphic" values="
+						0.10889, 0.89111, 0, 0, 0
+						0.10889, 0.89111, 0, 0, 0
+						0.00447, -0.00447, 1, 0, 0
+						0, 0, 0, 1, 0"
+					/>
+				</filter>
+			</svg>
+		`;
+	}
+
+	_renderControls() {
+		if (this.hideControls) return;
+
+		const colorblindTypes = Object.entries(DISABILITY_TYPES).filter(value => value[0].includes('colorblind'));
+		const otherTypes = Object.entries(DISABILITY_TYPES).filter(value => !value[0].includes('colorblind'));
```

Comment:
These filters would end up getting run every time `_renderControls` is called but would return the same value each time. Would it be possible to restructure this such that this uses some sort of constant?